Modularization of files and nice details implemented

fcorencoret opened the pull request:
Separated files within their scope (utils, dataloaders, models, parsers) and implemented nice training details as:

- tqdm progress bar for each pruning_iter - showing the accuracy of the last epoch during that iteration
- DataParallel to train models several GPUs

fcorencoret committed:
Modularizing files and added tqdm for training

fcorencoret committed:
Implemented DataParallel for faster training

sdeepaknarayanan commented:
Hi @fcorencoret, thank you for your interest in our repository. 

We are currently busy but hope to get back to this PR ASAP.

fcorencoret committed:
Implemented new features :rocket:

Implemented training with freezing the lottery ticket masks, training with the convolution layers freezed, traininig with random weights

sdeepaknarayanan commented:
Hi @fcorencoret , we will merge your PR to a new branch dev. We would like to keep master intact for now.
